src/components/entity/table.vue

```diff
@@ -76,23 +77,66 @@ const emit = defineEmits(['update', 'resolve', 'delete', 'restore', 'selectionCh
 
 // The component does not assume that this data will exist when the component is
 // created.
-const { project, odataEntities } = useRequestData();
+const { project, dataset, odataEntities } = useRequestData();
 
-const afterAction = ({ target, data, index }) => {
+const afterAction = ({ target, data }) => {
   const { classList } = target;
+  if (target.classList.contains('restore-button'))
+    emit('restore', data);
+
+  const restEntity = odataEntityToRest(data, dataset.properties);
   if (classList.contains('delete-button'))
-    emit('delete', data);
+    emit('delete', restEntity);
   else if (classList.contains('update-button'))
-    emit('update', index);
+    emit('update', restEntity);
   else if (classList.contains('resolve-button'))
-    emit('resolve', index);
-  else if (target.classList.contains('restore-button'))
-    emit('restore', data);
+    emit('resolve', restEntity);
 };
 const table = ref(null);
-const afterUpdate = (index) => { markRowsChanged(table.value.getRowPair(index)); };
-const afterDelete = (index) => { markRowsDeleted(table.value.getRowPair(index)); };
-defineExpose({ afterUpdate, afterDelete });
+const findIndex = (uuid) =>
+  odataEntities.value.findIndex(entity => entity.__id === uuid);
+const afterUpdate = (updatedEntity, resolving) => {
+  const index = findIndex(updatedEntity.uuid);
+
+  // Update the OData using the REST response.
+  const oldOData = odataEntities.value[index];
+  const newOData = Object.assign(Object.create(null), {
+    __id: oldOData.__id,
+    label: updatedEntity.currentVersion.label,
+    __system: {
+      ...oldOData.__system,
+      version: updatedEntity.currentVersion.version,
+      updates: oldOData.__system.updates + 1,
+      updatedAt: updatedEntity.updatedAt
+    }
+  });
+  const { data: updatedData } = updatedEntity.currentVersion;
+  for (const { name, odataName } of dataset.properties)
+    newOData[odataName] = updatedData[name];
+  odataEntities.value[index] = newOData;
+
+  if (!resolving) markRowsChanged(table.value.getRowPair(index));
```

Comment:
Here's a different option for passing the updated data to the resolve modal:

```suggestion
  if (!resolving) {
    markRowsChanged(table.value.getRowPair(index));
    emit('resolve', newOData);
  }
```